src/model/model.nit

```diff
@@ -355,14 +355,19 @@ private class MPropDefSorter
 	super Comparator
 	redef type COMPARED: MPropDef
 	var mmodule: MModule
+
+	private var mclassdef_sorter: MClassDefSorter is noinit
+
+	init
+	do
+		mclassdef_sorter = new MClassDefSorter(mmodule)
```

Comment:
maybe expose `mclassdef_sorter` and `mpropdef_sorter` as services (attributes?) of MModule? That should avoid this complex construction.